### ISSUE TITLE
JSDoc Class extending Array not supported (?)

### ISSUE THREAD

**Issue by kinsi55** [microsoft/TypeScript#30176]
Got a Class that Extends Array (FooArray) and am trying to create a Typed FooArray in another place. JSDoc suggests that I would be correct by using `FooArray.<Type>` (As the "normal" suggested method is either `Type[]` or `Array.<Type>`) and this markdown indeed seems to be parsed by jsdoc without issues, however VSC ends up detecting neither `FooArray` nor `Type`

Full example:

```
/**
 * @class FooArray
 * @extends {Array}
 */
class FooArray extends Array {
	constructor() {
		super();
		//whatever
	}


	/**
	 * returns "baz"
	 * @returns {string}
	 */
	bar() {
		return "baz";
	}
};

//This array will be filled with only values of type XYZ

/** @type {Array.<XYZ>} => instance[0] is correctly tagged / detected, instance.bar() is not (Obviously) */
/** @type {FooArray} => instance.bar() is documented but JSDoc has no idea of the values in the Array */
/** @type {FooArray.<XYZ>} => Neither is detected / documented */
let instance = new FooArray();
```

**Comment by RyanCavanaugh** [comment]
`FooArray.<XYZ>` cannot work with the declaration shown because `FooArray` has no declared type parameter, and `@extends {Array}` also omits `Array`'s required element type.

Declare the element type on the class and apply it to the base class:

```js
/** @typedef {{ kind: "xyz" }} XYZ */

/**
 * @class FooArray
 * @template T
 * @extends {Array<T>}
 */
class FooArray extends Array {
    /** @returns {string} */
    bar() {
        return "baz";
    }
}

/** @type {FooArray.<XYZ>} */
const instance = new FooArray();
```

With `--allowJs --checkJs`, this gives `instance[0]` type `XYZ` and `instance.bar()` type `string`. TypeScript 3.3.3333 and native 7.1.0-dev.20260901.1 both reject assigning those expressions to `number`.
